modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/impl/ExtensionPointImpl.java

```diff
@@ -56,6 +66,15 @@
     @XParent
     public RegistrationInfo ri;
 
+    // potential registry class declaration
+    @XNode(value = "registry@class")
+    protected String registryKlass;
+
+    // final operating registry class
+    protected Registry registry;
```

Comment:
Could we have it outside of this class, like in `ComponentManager`? The reason is that this object should be a POJO imho.

Reply:
Ok i'll try to move it when reviewing all commits

Reply:
Note that this means moving the registration logics out of this class (and similarly, the registry retrieval will trigger a lookup to the component manager).
Note that for compat logics, i also think i should stop loading contribution objects on the Extension objects (that should be a POJO too if we're consistent) when a registry is defined: nobody should expect the extension to hold contributions anymore when using registries, and this will spare unneeded xmap load of these objects, wdyt?

Reply:
done with #4537